cornflow/tests/custom_test_case.py

```diff
@@ -426,3 +427,189 @@ def test_payload_bad_format(self):
+
+        def test_old_token(self):
+            token = (
+                "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE2MTA1MzYwNjUsImlhdCI6MTYxMDQ0OTY2NSwic3ViIjoxfQ"
+                ".QEfmO-hh55PjtecnJ1RJT3aW2brGLadkg5ClH9yrRnc "
+            )
+
+            payload = self.data
+
+            response = self.client.post(
+                LOGIN_URL,
+                data=json.dumps(payload),
+                follow_redirects=True,
+                headers={"Content-Type": "application/json"},
+            )
+
+            self.id = response.json["id"]
+
+            response = self.client.get(
+                USER_URL + str(self.id) + "/",
+                follow_redirects=True,
+                headers={
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer " + token,
+                },
+            )
+
+            self.assertEqual(400, response.status_code)
+            self.assertEqual(
+                "Token expired, please login again", response.json["error"]
+            )
+
+        def test_bad_format_token(self):
+            response = self.client.post(
+                LOGIN_URL,
+                data=json.dumps(self.data),
+                follow_redirects=True,
+                headers={"Content-Type": "application/json"},
+            )
+
+            token = response.json["token"]
+            self.id = response.json["id"]
+
+            response = self.client.get(
+                USER_URL + str(self.id) + "/",
+                follow_redirects=True,
+                headers={
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer" + token,
+                },
+            )
+            self.assertEqual(400, response.status_code)
+
+        def test_invalid_token(self):
+            token = (
+                "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE2MTA1Mzk5NTMsImlhdCI6MTYxMDQ1MzU1Mywic3ViIjoxfQ"
+                ".g3Gh7k7twXZ4K2MnQpgpSr76Sl9VX6TkDWusX5YzImo"
+            )
+
+            payload = self.data
+
+            response = self.client.post(
+                LOGIN_URL,
+                data=json.dumps(payload),
+                follow_redirects=True,
+                headers={"Content-Type": "application/json"},
+            )
+            self.id = response.json["id"]
+
+            response = self.client.get(
+                USER_URL + str(self.id) + "/",
+                follow_redirects=True,
+                headers={
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer " + token,
+                },
+            )
+
+            self.assertEqual(400, response.status_code)
+            self.assertEqual(
+                "Invalid token, please try again with a new token",
+                response.json["error"],
+            )
+
+        def test_token(self):
+            # TODO: implement to check correct token creation
+            pass
```

Comment:
this is missing?

Reply:
Is a test that is dificult to implement as the token has the timestamp on it, so I do not know how to test that I can generate a token that is the same that is generated at log in time.